src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -27,6 +30,35 @@
      */
     private static final Pattern BASIC_COMMAND_FORMAT = Pattern.compile("(?<commandWord>\\S+)(?<arguments>.*)");
 
+    /**
+     * Used to store the last n commands for easy reference.
+     */
+    private static LinkedList<String> previousCommands = new LinkedList<>();
+
+
+    //TODO: Change this to non static methods
+    /**
+     * Shows the last used commands to the user only if it is not u
+     *
+     * @return a String of the past commands
+     */
+    public static String getPreviousCommands() {
+        if (previousCommands == null || previousCommands.size() == 0) {
+            return "";
+        }
+        return previousCommands.poll();
+    }
+
+    /**
+     * Keeps track of Commands inputted.
+     */
+    public static void addCommand(String command) {
+        if (command.equals(PREVIOUS_COMMAND_WORD)) {
```

Comment:
The guard clause here might not be helpful in this case because inputs such as "u   " etc might break the program.